.github/workflows/ci-vulnerability-scans.yml

```diff
@@ -0,0 +1,81 @@
+# This file is an example CI test pipeline that will run vulnerability scans on your image
+# to ensure images built are secure before they are deployed. 
+name: API Linting & Testing CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  check-for-changes:
+    runs-on: ubuntu-latest
+    # Set job outputs to values from filter step
+    outputs:
+      needs-testing: ${{ steps.filter.outputs.needs-testing }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            needs-testing:
+              - app/**
+              - .github/**/*.yml
+
+  image-scan:
+    runs-on: ubuntu-latest
+    needs: check-for-changes
+    if: ${{ needs.check-for-changes.outputs.needs-testing == 'true' }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      # Scans Dockerfile for any bad practices or issues before building image
+      - name: Scan Dockerfile by hadolint
+        uses: hadolint/hadolint-action@v3.0.0
+        with:
+          dockerfile: app/Dockerfile
+          format: tty
+          failure-threshold: warning
+
+      - name: Build and tag Docker image for scanning
+        id: build-image
+        run: |
+          make release-build
+          echo "image=template-infra-app:latest" >> $GITHUB_OUTPUT
+
+      - name: Scan server image
+        uses: ./.github/actions/image-scan
+        with:
+          image: ${{ steps.build-image.outputs.image }}
+
+  linting-and-testing:
+    runs-on: ubuntu-latest
+    needs: check-for-changes
+    if: ${{ needs.check-for-changes.outputs.needs-testing == 'true' }}
+    defaults:
+      run:
+        working-directory: ./app
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Initialize the docker containers
+        run: make init
+
+      # Change this step to your testing make commands, you can run a list here
+      - name: Run Tests
+        run: make testing
+
+  lint-test:
+    # Orchestration job that runs the other jobs in parallel
+    runs-on: ubuntu-latest
+    needs:
+      - check-for-changes
+      - linting-and-testing
+      - image-scan
+    if: ${{ needs.check-for-changes.outputs.needs-testing == 'true' }}
+    steps:
+      - run: echo "Finished running tests successfully!"
```

Comment:
Let's remove application testing and linting. Those should be implemented in individual application repos in `ci-app.yml` (see https://github.com/navapbc/template-application-flask/blob/main/.github/workflows/ci-app.yml and https://github.com/navapbc/template-application-nextjs/blob/main/.github/workflows/ci-app.yml) and is testing the application code rather than the infrastructure

Reply:
Should I remove it here and make a template that has it to show the process?

Reply:
What did you have in mind? template-application-flask and template-application-nextjs both have it already, so anyone using the template application would already have an example application linting and testing. Were you thinking of a third template or something else?

Reply:
For a future PR, make a third one with them all in it. I was thinking if we are scanning an image that the linting and testing jobs might be running against, we could share the image so we're building it once. Obviously can't do that for this PR since we haven't been able to get it working. 

Also, to maybe make things somewhat cleaner by needing a single job compared to multiple jobs for merging. We have our repo rules to not allow merges from PRs until the orchestration job is successful, so having a single job call all the other jobs would make it somewhat easier than needing to have multiple files that have multiple blocking jobs required for merges

Reply:
Oh, in terms of example repos that integrate template-infra and template-application and have all of the CI jobs running, we have [platform-test](https://github.com/navapbc/platform-test) and [platform-test-nextjs](https://github.com/navapbc/platform-test-nextjs). I haven't yet created platform-test-flask but it's in the backlog.

Reply:
> Also, to maybe make things somewhat cleaner by needing a single job compared to multiple jobs for merging. We have our repo rules to not allow merges from PRs until the orchestration job is successful, so having a single job call all the other jobs would make it somewhat easier than needing to have multiple files that have multiple blocking jobs required for merges

I don't think putting all the checks in one job is cleaner. I think it's cleaner to have the checks separate, so that you have a sense of what is failing just by looking at which check failed, rather than having to click into the check.

Reply:
For you first comment, yes something like that. For the second one, I was meaning something along the lines of this

- Branch protection rule requires job named `image-scans` to be successful
- That job runs all other jobs in parallel (as in this action in the PR)
- Any fails will fail the overall job, where you can click into that (see below in the checks)
- Any new jobs that we might want to add in the future would just be added to the orchestration job to run

I'm saying this because in NJ, our infra is in a separate repo, so any changes like required checks need a different PR to do it

Reply:
Maybe I'm misunderstanding, but how is changing required checks related to infrastructure? Isn't that just Github setting that you'd change through the UI (see screenshot below)? Or are y'all managing github repo settings via code somehow?

<img width="784" alt="image" src="https://user-images.githubusercontent.com/447859/214186214-552b4470-6f27-4bec-a486-0c24e8ad3d63.png">

Reply:
Yeah, we managed the settings with Terraform. There's a couple things that were changed manually since I deployed the settings, and I need to update our statefile with that in case we ever need to do a redeploy or copy the settings to another repo for another one of our teams on NJ